lib/addons/aws-privateca-issuer/index.ts

```diff
@@ -0,0 +1,103 @@
+// lib/kubevious_addon.ts
+import { Construct } from 'constructs';
+import { ManagedPolicy } from "aws-cdk-lib/aws-iam";
+import merge from "ts-deepmerge";
+import { ServiceAccount } from 'aws-cdk-lib/aws-eks';
+import { HelmAddOn, HelmAddOnUserProps, HelmAddOnProps } from "../helm-addon";
+import { dependable, setPath, createNamespace } from '../../utils';
+import { ClusterInfo, Values } from "../../spi";
+
+/**
+ * User provided options for the Helm Chart
+ */
+export interface AWSPrivateCAIssuerAddonProps extends HelmAddOnUserProps {
+
+    /**
+     * The name of the service account to use. If createServiceAccount is true, a serviceAccountName is generated.
+     */
+    serviceAccountName?: string;
+    /**
+     * An array of Managed IAM Policies which Service Account needs for IRSA Eg: irsaRoles:["AWSCertificateManagerPrivateCAFullAccess"]. If not empty
+     * Service Account will be Created by CDK with IAM Roles Mapped (IRSA). In case if its empty, Service Account will be created with out IAM Roles
+     */   
+     irsaRoles?: string[];    
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps: HelmAddOnProps & AWSPrivateCAIssuerAddonProps = {
+  name: "blueprints-aws-pca-issuer-addon",
+  chart: "aws-privateca-issuer",
+  namespace:"aws-pca-issuer",
+  version: "1.2.2",
+  release: "aws-pca-issuer",
+  repository:  "https://cert-manager.github.io/aws-privateca-issuer",
+  values: {},
+  serviceAccountName: "aws-pca-issuer",
+  irsaRoles: []

```

Comment:
I think it would be beneficial to add the AWSCertificateManagerPrivateCAFullAccess by default and allow customers to override. If they pass policies, we use those. If they don't we apply the default which can be used to instantiate the add-on with no parameters.

Reply:
done